src/base/utils/misc.cpp

```diff
@@ -345,30 +370,34 @@ bool Utils::Misc::isPreviewable(const QString &extension)
 
 // Take a number of seconds and return an user-friendly
 // time duration like "1d 2h 10m".
-QString Utils::Misc::userFriendlyDuration(qlonglong seconds)
+QString Utils::Misc::userFriendlyDuration(std::chrono::seconds seconds)
 {
-    if ((seconds < 0) || (seconds >= MAX_ETA))
+    using std::chrono::duration_cast;
+    using namespace std::chrono_literals;
+
+    if (seconds >= MAX_ETA)
         return QString::fromUtf8(C_INFINITY);
 
-    if (seconds == 0)
+    if (seconds.count() == 0)
         return "0";
 
-    if (seconds < 60)
+    if (seconds < 1min)
         return QCoreApplication::translate("misc", "< 1m", "< 1 minute");
 
-    qlonglong minutes = seconds / 60;
-    if (minutes < 60)
-        return QCoreApplication::translate("misc", "%1m", "e.g: 10minutes").arg(QString::number(minutes));
+    auto minutes = std::chrono::floor<std::chrono::minutes>(seconds);
+    if (minutes < 1h)
+        return QCoreApplication::translate("misc", "%1m", "e.g: 10minutes").arg(QString::number(minutes.count()));
 
-    qlonglong hours = minutes / 60;
-    minutes -= hours * 60;
-    if (hours < 24)
-        return QCoreApplication::translate("misc", "%1h %2m", "e.g: 3hours 5minutes").arg(QString::number(hours), QString::number(minutes));
+    auto hours = duration_cast<std::chrono::hours>(minutes);
+    minutes -= hours;
+    if (hours.count() < 24)
+        return QCoreApplication::translate("misc", "%1h %2m", "e.g: 3hours 5minutes")
+            .arg(QString::number(hours.count()), QString::number(minutes.count()));
 
-    qlonglong days = hours / 24;
-    hours -= days * 24;
+    auto days = hours.count() / 24;
+    hours -= std::chrono::hours(days * 24);
```

Comment:
duration has `operator%` which may be simpler, or at least more straight forward

Reply:
But `std::chrono::days` is from C++20. Or did I interpret the suggestion incorrectly?